compose/project.py

```diff
@@ -602,6 +602,12 @@ def _get_convergence_plans(self, services, strategy, always_recreate_deps=False)
     def pull(self, service_names=None, ignore_pull_failures=False, parallel_pull=False, silent=False,
              include_deps=False):
         services = self.get_services(service_names, include_deps)
+        can_build_images = [service.image_name for service in filter(
+            lambda service: service.can_be_built(), services)]
+
+        can_pull = filter(lambda service: not service.can_be_built()
```

Comment:
If a service can be built, it will already be in `can_build_images`, so we can simplify the boolean condition.

I suggest the following (using `images_to_build` in place of `can_build_images`:
```python
services_to_pull = [service for service in services if service.image_name not in images_to_build]
```